packages/boot/test/bootstrapTests/ec-contract-upgrade.test.ts

```diff
@@ -0,0 +1,645 @@
+import { test as anyTest } from '@agoric/zoe/tools/prepare-test-env-ava.js';
+import { TestFn } from 'ava';
+import { eventLoopIteration } from '@agoric/internal/src/testing-utils.js';
+import {
+  makeAgoricNamesRemotesFromFakeStorage,
+  slotToBoardRemote,
+  unmarshalFromVstorage,
+} from '@agoric/vats/tools/board-utils.js';
+import { makeMarshal } from '@endo/marshal';
+
+import { makeSwingsetTestKit } from '../../tools/supports.js';
+import { makeWalletFactoryDriver } from '../../tools/drivers.js';
+
+const wallets = {
+  gov1: 'agoric1gx9uu7y6c90rqruhesae2t7c2vlw4uyyxlqxrx',
+  gov2: 'agoric1d4228cvelf8tj65f4h7n2td90sscavln2283h5',
+  gov3: 'agoric14543m33dr28x7qhwc558hzlj9szwhzwzpcmw6a',
+  gov4: 'agoric13p9adwk0na5npfq64g22l6xucvqdmu3xqe70wq',
+  gov5: 'agoric1el6zqs8ggctj5vwyukyk4fh50wcpdpwgugd5l5',
+  gov6: 'agoric1zayxg4e9vd0es9c9jlpt36qtth255txjp6a8yc',
+};
+
+const latestQuestionKey = `published.committees.Economic_Committee.latestQuestion`;
+const lastOutcomeKey = `published.committees.Economic_Committee.latestOutcome`;
+const highPrioritySenderKey = 'highPrioritySenders';
+
+const offerIds = {
+  propose: { outgoing: 'outgoing_propose' },
+  vote: { outgoing: 'outgoing_vote', incoming: 'incoming_vote' },
+};
+
+const getQuestionId = id => `propose-question-${id}`;
+const getVoteId = id => `vote-${id}`;
+
+export const makeZoeTestContext = async t => {
+  console.time('ZoeTestContext');
+  const swingsetTestKit = await makeSwingsetTestKit(t.log, undefined, {
+    configSpecifier: '@agoric/vm-config/decentral-main-vaults-config.json',
+  });
+
+  const { runUtils, storage } = swingsetTestKit;
+  console.timeLog('DefaultTestContext', 'swingsetTestKit');
+  const { EV } = runUtils;
+
+  await eventLoopIteration();
+
+  // We don't need vaults, but this gets the brand, which is checked somewhere
+  // Wait for ATOM to make it into agoricNames
+  await EV.vat('bootstrap').consumeItem('vaultFactoryKit');
+  console.timeLog('DefaultTestContext', 'vaultFactoryKit');
+
+  // has to be late enough for agoricNames data to have been published
+  const agoricNamesRemotes = makeAgoricNamesRemotesFromFakeStorage(storage);
+  console.timeLog('DefaultTestContext', 'agoricNamesRemotes');
+
+  console.timeEnd('DefaultTestContext');
+  const walletFactoryDriver = await makeWalletFactoryDriver(
+    runUtils,
+    storage,
+    agoricNamesRemotes,
+  );
+
+  const smartWallets = await Promise.all(
+    Object.values(wallets).map(async addr =>
+      walletFactoryDriver.provideSmartWallet(addr),
+    ),
+  );
+  const { fromCapData } = makeMarshal(undefined, slotToBoardRemote);
+
+  const getUpdatedDebtLimit = () => {
+    const atomGovernance = unmarshalFromVstorage(
+      storage.data,
+      'published.vaultFactory.managers.manager0.governance',
+      fromCapData,
+      -1,
+    );
+    return atomGovernance.current.DebtLimit.value.value;
+  };
```

Comment:
This sort of helper function makes the actual test much easier to read. Thanks.

I'd like to see more of this.

Consider organizing the code so that EC members are represented as objects created with `makeECMember(wallet, voterId, agoricNamesRemotes.instance)`. Then we can shorten

```
      return w.executeOffer({
        id: offerIds.vote.outgoing,
        invitationSpec: {
          source: 'purse',
          instance: economicCommittee,
          description: invitation.description,
        },
        proposal: {},
      });
```

down to something like `members.gov1.acceptCommitteeInvitation()`.

Reply:
The `governanceDriver` would do pretty much this.